keps/0002-controller-manager.md

```diff
@@ -254,7 +258,100 @@ In case of the cloud-controller-manager, the deployment should be deleted using
 kubectl delete -f cloud-controller-manager.yml
 ```
 
-### Implementation Details/Notes/Constraints [optional]
+### Implementation Details/Notes/Constraints
+
+#### Repository Requirements
+
+**This is a proposed structure, and may change during the 1.11 release cycle.
+WG-Cloud-Provider will work with individual sigs to refine these requirements
+to maintain consistency while meeting the technical needs of the provider
+maintainers**
+
+Each cloud provider hosted within the `kubernetes` organization shall have a
+single repository named `kubernetes/cloud-provider-<provider_name>`. Those
+repositories shall have the following structure:
+
+* A `cloud-controller-manager` subdirectory that contains the implementation
+  of the provider-specific cloud controller.
+* A `docs` subdirectory.
+* A `docs/cloud-controller-manager.md` file that describes the options and
+  usage of the cloud controller manager code.
+* A `tests` subdirectory that contains testing code.
```

Comment:
I don't think this should be a requirement. Although this is what we do for the e2e tests for Kubernetes. Some reasonable people think that having tests moved away from the code is an antipattern. Perhaps we should require a `TESTING.md` or similar which specifies how to run the tests for humans and a `Makefile` as a test entrypoint